corehq/apps/reports/v2/filters/xpath_column.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from collections import namedtuple
+
+from django.utils.translation import ugettext_lazy
+
+from corehq.apps.reports.v2.models import BaseFilter
+from corehq.apps.reports.v2.exceptions import ColumnFilterNotFound
+
+
+ChoiceMeta = namedtuple('ChoiceMeta', 'title name operator')
+AppliedFilterContext = namedtuple(
+    'AppliedFilterContext',
+    'filter_name choice_name value'
+)
+
+
+class DataType(object):
+    TEXT = 'text'
+    NUMERIC = 'numeric'
+
+
+class Clause(object):
+    ALL = 'all'
+    ANY = 'any'
+
+
+class BaseXpathColumnFilter(BaseFilter):
+    name = None
+    title = None
+    data_type = None
+    choices = []
+
+    @classmethod
+    def get_expression(cls, property, choice_name, value):
+        value = cls.format_value(value)
+        operator = cls.get_operator(choice_name)
+        return "{property} {operator} {value}".format(
+            property=property,
+            operator=operator,
+            value=value,
+        )
+
+    @classmethod
+    def format_value(cls, value):
+        return value
+
+    @classmethod
+    def get_operator(cls, choice_name):
+        name_to_operator = {c.name: c.operator for c in cls.choices}
+        # Let this fail hard on KeyError, as it signals an issue on the JS side
+        return name_to_operator[choice_name]
+
+    @classmethod
+    def get_context(cls):
+        return {
+            'title': cls.title,
+            'name': cls.name,
+            'choices': [
+                {
+                    'title': c.title,
+                    'name': c.name
+                } for c in cls.choices
+            ],
+        }
+
+
+class TextXpathColumnFilter(BaseXpathColumnFilter):
+    title = ugettext_lazy("Text")
+    name = 'xpath_column_text'
+    data_type = DataType.TEXT
+    choices = [
+        ChoiceMeta(ugettext_lazy("Equals"), 'text_equals', '='),
+        ChoiceMeta(ugettext_lazy("Does not equal"), 'text_not_equals', '!='),
+    ]
+
+    @classmethod
+    def format_value(cls, value):
+        return "'{}'".format(value)
+
+
+class NumericXpathColumnFilter(BaseXpathColumnFilter):
+    title = ugettext_lazy("Number")
+    name = 'xpath_column_number'
+    data_type = DataType.NUMERIC
+    choices = [
+        ChoiceMeta(ugettext_lazy("Is equal to"), 'num_equals', '='),
+        ChoiceMeta(ugettext_lazy("Is less than"), 'num_less_than', '<'),
+        ChoiceMeta(ugettext_lazy("Is greater than"), 'num_greater_than', '>'),
+        ChoiceMeta(ugettext_lazy("Is not equal to"), 'num_not_equals', '!='),
+    ]
+
+    @staticmethod
+    def _get_number(value):
+        try:
+            return int(value)
```

Comment:
no it's not because if it's passed a string that is actually float it will throw a value error.

```
In [1]: int('1.0')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-7d3564d09835> in <module>()
----> 1 int('1.0')

ValueError: invalid literal for int() with base 10: '1.0'
```

Reply:
Interesting. I do think it is still worth catching the potential ValueError from `float('foobar')` here, but will leave to your judgement.